library/core/src/ptr/non_null.rs

```diff
@@ -117,6 +117,25 @@ impl<T: ?Sized> NonNull<T> {
     /// The resulting lifetime is bound to self so this behaves "as if"
     /// it were actually an instance of T that is getting borrowed. If a longer
     /// (unbound) lifetime is needed, use `&*my_ptr.as_ptr()`.
+    ///
+    /// # Safety
+    ///
+    /// When calling this method, you have to ensure that *either* the pointer is NULL *or*
+    /// all of the following is true:
```

Comment:
This is a `NonNull` pointer. It can never be NULL. So why the "either the pointer is NULL" part?

Reply:
Good catch! That part was left over from the documentation of the Pointer `as_{ref,mut}` methods. Should be fixed now.